hr_expense_petty_cash/models/account_move.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import float_compare
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    is_petty_cash = fields.Boolean(
+        string="Petty Cash",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+
+    def action_post(self):
+        self._check_petty_cash_amount()
+        return super().action_post()
+
+    @api.constrains("invoice_line_ids", "line_ids")
+    def _check_petty_cash_amount(self):
+        petty_cash_env = self.env["petty.cash"]
```

Comment:
```suggestion
        petty_cash_env = self.env["petty.cash"].sudo()
```
Please add sudo() on petty.cash model. This function is affect with another module such as purchase module.

I have to create vendor bill on purchase order by procurement dept (Not be in accounting group), it's raise permission error.
![Selection_222](https://user-images.githubusercontent.com/24691983/131832123-908f4996-7310-41cb-9018-f52e8f472f2d.png)

Reply:
Thank you for your suggestion. Updated.